digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -203,4 +212,33 @@ class RoadAddressService(roadLinkService: RoadLinkService) {
       case ex: NumberFormatException => NoClass
     }
   }
+
+  def createMissingRoadAddress(missingRoadLinks: Seq[RoadAddressLink]) = {
+    withDynTransaction {
+      missingRoadLinks.map { links =>
+          val anomalyCode = getAnomalyCodeByLinkId(links.linkId, links.roadPartNumber)
+          RoadAddressDAO.createMissingRoadAddress(links.linkId, links.startAddressM, links.endAddressM, links.roadNumber, links.roadPartNumber, anomalyCode)
+      }
+    }
+  }
+
+  def getAnomalyCodeByLinkId(linkId: Long, roadPart: Long) : Int = {
+    var code = 0
+      //roadlink dont have road address
+      if (RoadAddressDAO.getLrmPositionByLinkId(linkId).length < 1) {
+        code = 1
+      }
+      //road address do not cover the whole length of the link
+       else if (RoadAddressDAO.getLrmPositionMeasures(linkId).length > 0)
```

Comment:
This verification shouldn't be done with the lenght of the road geometry and the length of the road address? Instead of comparing the start and end measure of the LrmPosition?